strings-csharp/HexToByteArray/HexToByteArray/ConvertHex.cs

```diff
@@ -16,5 +18,157 @@ public static byte[] FromHexStringAlternative(ReadOnlySpan<char> hex)
                     System.Globalization.NumberStyles.HexNumber);
             return bytes;
         }
+
+        public static byte[] FromHexWithPointers(string hex)
+        {
+            if (hex.Length % 2 != 0) throw new ArgumentException("Invalid hex string", nameof(hex));
+
+            var res = new byte[hex.Length / 2]; // Allocate memory
```

Comment:
In the case of these examples, we are always going to set each value of the array, so it is more performant to use [GC.AllocateUninitializedArray<T>](https://learn.microsoft.com/en-us/dotnet/api/system.gc.allocateuninitializedarray). This creates an unarray but it does not perform the step to "zero" it, thus saving a few instructions. Also, instead of doing the division we can use a left shift of 1 to divide the number in half. This generates a shift instruction instead of a division instruction.